internal/cmd/io_binary.go

```diff
@@ -105,6 +107,34 @@ func NewBinaryIO(ctx context.Context, id string, uri *url.URL) (_ UpstreamIO, er
 	}, nil
 }
 
+// sanitizePath parses the URL object and returns an absolute path to the logging driver
+func sanitizePath(uri *url.URL) (sanitized string) {
+	path := uri.Path
+	// Convert windows-style slashes to linux-style to clean the path
+	path = strings.Replace(path, "\\", "/", -1)
+	path = filepath.Clean(path)
+	// Path is clean, convert back
+	sanitized = strings.Replace(path, "/", "\\", -1)
+	if sanitized == "." {
+		sanitized = ""
+	}
+
+	// Strip any leading '\\' if the path contains a hard drive in it.
+	if strings.Contains(sanitized, ":\\") {
+		if sanitized[0] == '\\' {
+			// For cleaned paths there should be only one leading '\\'
+			return sanitized[1:]
+		}
+		return sanitized
+	}
+
+	// Absolute path is required, treat "binary://path/to/binary" and "binary:///path/to/binary"
+	// as the same.
+	if uri.Host != "" {
+		return "\\" + uri.Host + sanitized
+	}
```

Comment:
Is it possible that uri.Host could already have a leading `\\` like in the hard drive case above?

Reply:
`url.Parse` will complain about back slashes in hostname, so things should fail before we even get here.